### PR TITLE
Radio Backpack

### PR DESCRIPTION
**Pull request by EvilMelonTheSecond**
## About The Pull Request
Adds a Radio Backpack to engineer vendor that can access req requests console and function like an antenna

## Why It's Good For The Game
I hate the disconnect between shipside req and groundside engineers who are supposed to work together to supply the entire team with gear. This backpack doesn't add anything new to the game, its simply an easier way to access existing mechanics without having to rely comms which tends to be excessively cluttered. I hope the ease of access to req will push engineers to capture more miners and give the game some more objective focus.

If this PR goes into testing and turns out well I have some other ideas like giving the radiopack an auto request approve timer if RO doesn't deny the order. Maybe the increased req activity will give a clearer picture for future miner rebalancing too.

## Changelog
:cl:
add: Added an engi backpack for groundside supply requests
/:cl:

### CODE REVIEW

**Comment by DarkWolfKai** [comment]
alternative name : RTO's long gone cousin

**Comment by bowlofcereal** [comment]
alternative title: makes the RO wake the fuck up

**Comment by ilikesaying** [comment]
Alternative Title "WHERE THE FUCK IS MY FUCKING ORDER?"

**Comment by Tristan63** [comment]
spam the RO with SADAR orders like none other before

**Review by EvilMelonTheSecond — COMMENTED** [review]
kill me

File: code/modules/reqs/supply.dm [hunk 1]
```diff
@@ -622,3 +622,43 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 	if(!SU)
 		SU = new(src)
 	return SU.interact(user)
+
+/obj/item/storage/backpack/marine/radiopack
+	name = "\improper TGMC radio operator backpack"
+	desc = "A backpack that resembles the ones old-age radio operator soldiers would use."
+	icon_state = "radiopack"
+	var/datum/supply_ui/requests/SU
```
Comment (BraveMole) [inline code/modules/reqs/supply.dm:630]: better var name

File: code/modules/reqs/supply.dm [hunk 2]
```diff
@@ -622,3 +622,43 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
...
+	var/datum/supply_beacon/beacon_datum
```
Comment (BraveMole) [inline code/modules/reqs/supply.dm:631]: both vars need autodoc

File: code/modules/reqs/supply.dm [hunk 3]
```diff
@@ -622,3 +622,43 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
...
+	if(.)
+		return
+	if(!allowed(user))
+		return
+	if(!SU)
+		SU = new(src)
+	return SU.interact(user)
+
+/obj/item/storage/backpack/marine/radiopack/attack_self(mob/living/user)
+	var/turf/location = get_turf(src)
```
Comment (BraveMole) [inline code/modules/reqs/supply.dm:644]: put that after all checks

File: code/modules/reqs/supply.dm [hunk 4]
```diff
@@ -622,3 +622,43 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
...
+	if(beacon_datum)
+		UnregisterSignal(beacon_datum, COMSIG_PARENT_QDELETING)
+		QDEL_NULL(beacon_datum)
+		user.show_message(span_warning("The [src] beeps and states, \"Your last position is no longer accessible by the supply console"), EMOTE_AUDIBLE, span_notice("The [src] vibrates but you can not hear it!"))
+		return
+	if(!is_ground_level(user.z))
+		to_chat(user, span_warning("You have to be on the planet to use this or it won't transmit."))
+		return FALSE
+	var/area/A = get_area(user)
+	if(A && istype(A) && A.ceiling >= CEILING_METAL)
```
Comment (BraveMole) [inline code/modules/reqs/supply.dm:654]:
```suggestion
	if(A?.ceiling >= CEILING_METAL)
```

File: code/modules/reqs/supply.dm (hunk 2 above)
Comment (EvilMelonTheSecond) [reply code/modules/reqs/supply.dm]: I assume autodoc means a comment explaining what the var is. My coding lingo knowledge is very limited.

**Review by BraveMole — APPROVED** [review]

**Comment by Losenis** [comment]
Put it in the synthvendor backpack choices too

**Pull request merged**